infrastructure/oss/oss.go

```diff
@@ -238,10 +238,8 @@ func (oss *Scanner) unmarshallAndRetrieveAnalysis(ctx context.Context, res []byt
 		targetFileUri := uri.PathToUri(targetFilePath)
 		fileContent, err := os.ReadFile(targetFilePath)
 		if err != nil {
-			log.Err(err).Str("method", "unmarshallAndRetrieveAnalysis").
-				Msgf("Error while reading the file %v, err: %v", targetFile, err)
-			oss.errorReporter.CaptureErrorAndReportAsIssue(path, err)
-			return nil
+			// don't fail the scan if we can't read the file. No annotations with ranges, though.
+			fileContent = []byte{}
```

Comment:
these changes seem unrelated. What is the reason for them?

Reply:
True. LS was erroring when not having a defined targetFile, e.g. when scanning unmanaged. this should make it a bit more resilient - instead of erroring out, it now just skips range finding.

Reply:
do we have any tests that could cover this change?

Reply:
No yet, but I can write a test